core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2787,7 +2787,7 @@ private JoinBridgeManager<?> createLookupSourceFactoryManager(
 
             int taskConcurrency = getTaskConcurrency(session);
             OperatorFactory hashBuilderOperatorFactory;
-            if (useSpillingJoinOperator()) {
+            if (useSpillingJoinOperator(session)) {
```

Comment:
Can we rely on `spillEnabled` instead of `isSpillEnabled(session)` in `useSpillingJoinOperator` ?

Reply:
That make sense. I added another commit with that change